File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/DeviceMetric.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class DeviceMetric : DomainResource, IResource
{
    public required string Category { get; set; }
    public string? MeasurementPeriod { get; set; }
    public string? Color { get; set; }
    public Base.ResourceReference? Parent { get; set; }
    public Base.CodeableConcept? Unit { get; set; }
    public required Base.CodeableConcept Type { get; set; }
    public Base.ResourceReference? Source { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public DeviceMetric_Calibration[]? Calibration { get; set; }
    public string? OperationalStatus { get; set; }

    public class DeviceMetric_Calibration : BackboneElement
    {
        public string? Time { get; set; }
        public string? Type { get; set; }
        public string? State { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/DeviceMetric.py
from typing import Optional, List
from base import CodeableConcept
from base import Timing
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class DeviceMetric_Calibration(BackboneElement):
    time: Optional[str] = None
    type: Optional[str] = None
    state: Optional[str] = None

class DeviceMetric(DomainResource):
    category: str
    measurement_period: Optional[Timing] = None
    color: Optional[str] = None
    parent: Optional[Reference] = None
    unit: Optional[CodeableConcept] = None
    type: CodeableConcept
    source: Optional[Reference] = None
    identifier: Optional[List[Identifier]] = None
    calibration: Optional[List[DeviceMetric_Calibration]] = None
    operational_status: Optional[str] = None